src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -737,5 +743,35 @@ void GcsServer::PrintAsioStats() {
   }
 }
 
+void GcsServer::TryGlobalGC() {
+  if (cluster_task_manager_->GetPendingQueueSize() == 0) {
+    task_pending_schedule_detected_ = 0;
+    return;
+  }
+  // Trigger global gc to solve task pending.
+  // To avoid spurious triggers, only those after two consecutive
+  // detections and under throttling are sent out.
+  if (task_pending_schedule_detected_++ > 0 && global_gc_throttler_->AbleToRun()) {
+    rpc::ResourcesData resources_data;
+    resources_data.set_should_global_gc(true);
+
+    if (RayConfig::instance().use_ray_syncer()) {
+      syncer::RaySyncMessage msg;
+      msg.set_version(absl::GetCurrentTimeNanos());
+      msg.set_node_id(local_node_id_.Binary());
+      msg.set_message_type(syncer::MessageType::COMMANDS);
+      std::string serialized_msg;
+      RAY_CHECK(resources_data.SerializeToString(&serialized_msg));
+      msg.set_sync_message(std::move(serialized_msg));
+      ray_syncer_->OnDemandBroadcasting(msg);
```

Comment:
Why not use `ray_syncer_->BroadcastMessage(msg);` directly?

Reply:
Now `BroadcastMessage()` is a private function.